diff --git a/src/port.c b/src/port.c
index 0f94e16..277bb7d 100644
--- a/src/port.c
+++ b/src/port.c
@@ -415,9 +415,19 @@ int ortp_server_pipe_close(ortp_socket_t spipe){
 }
 
 ortp_socket_t ortp_client_pipe_connect(const char *name){
+	ortp_socket_t sock = -1;
 	struct sockaddr_un sa;
+	struct stat fstats;
 	char *pipename=make_pipe_name(name);
-	ortp_socket_t sock=socket(AF_UNIX,SOCK_STREAM,0);
+	uid_t uid = getuid();
+
+	// check that the creator of the pipe is us
+	if( (stat(name, &fstats) == 0) && (fstats.st_uid != uid) ){
+		ortp_error("UID of file %s (%u) differs from ours (%u)", pipename, fstats.st_uid, uid);
+		return -1;
+	}
+
+	sock = socket(AF_UNIX,SOCK_STREAM,0);
 	sa.sun_family=AF_UNIX;
 	strncpy(sa.sun_path,pipename,sizeof(sa.sun_path)-1);
 	ortp_free(pipename);
